File: sorting/bubble.py
# MEMBUAT ALGORITMA PROGRAM BUBBLE SORT DI PYTHON

def BubbleSortAscending (array) :
  N = len(array)
  for i in range(N) : # 0 - 9
    for j in range(0,  N - i - 1) : 
      if array[j] > array[j + 1] :
        temp = array[j]
        array[j] = array[j + 1]
        array[j + 1] = temp
  
  return array

ListNama = [1,4,2,5,7,3,6,9,8,10]

print(BubbleSortAscending(ListNama))

File: sorting/bubble.ts
const angkaAcak : number[] = [4,6,5,1,3,2,9,7,8,10,20,50,23,43,21,35];

const bubbleSortAscending = (array : number[]) : number[] => {
  const n:number = array.length;

  for(let i = 0; i < n - 1; i++) {
    for(let j = 0; j < n - 1 - i; j++) {
      if(array[j] > array[j + 1]) {
        let temp : number = array[j]
        array[j] = array[j + 1];
        array[j + 1] = temp;
      }
    }
  }

  return array;
}

console.log(bubbleSortAscending(angkaAcak));